apps/backend/modules/runner.py

```diff
@@ -79,98 +69,77 @@ def add_to_output_batch(fileOutput, ExpRun):
-            if trialExtraFile != '':
-                add_to_output_batch(trialExtraFile, 0)
-                firstTrial = OUTPUT_INDICATOR_USING_CSVS
-            if trialResult == '':
-                writer.writerow(["0", firstTrial] + get_configs_ordered(f'configFiles/{0}.ini', paramNames))
-            else:
-                if (output := get_output_results(trialResult)) is None:
-                    raise InternalTrialFailedError("Nothing returned when trying to get first non-header line of results (the first run?) (David, improve this error message please)")
-                writer.writerow(["0"] + output + get_configs_ordered(f'configFiles/{0}.ini', paramNames))
-            for i in range(1, numTrialsToRun + 1):
-                try:
-                    response_data = run_trial(filepath, f'configFiles/{i}.ini', filetype, i, keepLogs, trialTimeout)
-                except InternalTrialFailedError:
-                    print('The trial failed for some internal reason?')  # TODO should this halt all further experiment runs?
-                    fails += 1
-                    expRef.update({'fails': fails})
-                    continue
-                except TrialTimeoutError:
-                    print(f"Trial#{i} timed out")
-                    fails += 1
-                    expRef.update({'fails': fails})
-                    writer.writerow([i] + ["TIMEOUT" for i in range(numOutputs)] + get_configs_ordered(f'configFiles/{i}.ini', paramNames))
-                    continue
-
-                if trialExtraFile != '':
-                    response_data = OUTPUT_INDICATOR_USING_CSVS
-                    add_to_output_batch(trialExtraFile, i)
-                if trialResult == '':
-                    writer.writerow([i, response_data] + get_configs_ordered(f'configFiles/{i}.ini', paramNames))
-                else:
-                    output = get_output_results(trialResult)
-                    if output is None:
-                        raise InternalTrialFailedError("Nothing returned when trying to get first non-header line of results (the rest of the runs?) (David, improve this error message please)")
-                    writer.writerow([i] + output + get_configs_ordered(f'configFiles/{i}.ini', paramNames))
-                if response_data != PIPE_OUTPUT_ERROR_MESSAGE:
-                    passes += 1
-                    expRef.update({'passes': passes})
-                else:
-                    fails += 1
-                    expRef.update({'fails': fails})
+
+            if experiment.trialExtraFile != '':
+                add_to_output_batch(experiment.trialExtraFile, trialNum)
+
+            output = get_line_n_of_trial_results_csv(1, experiment.trialResult)
+            if output is None:
+                raise InternalTrialFailedError("Nothing returned when trying to get first non-header line of results (the rest of the runs?) (David, improve this error message please)")
```

Comment:
David, improve this error message please

Reply:
Changed to a try catch that re-raises the error from `get_line_n_of_trial_results_csv`